server/publications/collections/product.js

```diff
@@ -53,42 +54,56 @@ export function findProductMedia(publicationInstance, productIds) {
  */
 Meteor.publish("Product", function (productId) {
   check(productId, Match.OptionalOrNull(String));
+
   if (!productId) {
     Logger.debug("ignoring null request on Product subscription");
     return this.ready();
   }
+
+  // verify that parent shop is ready
+  const shop = Reaction.getCurrentShop();
+  if (typeof shop !== "object") {
+    return this.ready();
+  }
+
+
   let _id;
+  let productShopId;
+
+  // Default selector. Any changes come via specific roles
   const selector = {
     isDeleted: { $in: [null, false] },
     isVisible: true
   };
-  let productShopId;
-  const shop = Reaction.getCurrentShop();
-  // verify that parent shop is ready
-  if (typeof shop !== "object") {
-    return this.ready();
+
+  // Permits admins to view both visible and invisible products.
```

Comment:
Hi @spencern, there's already a role checking on line 132
```js 
if (Reaction.hasPermission("createProduct", this.userId, productShopId)) {
```
There's also no need for [`owner` or `admin`], as `createProduct` perm is strong enough to be together with any of those users' roles...
Leaving visible undeclared will just look to return `any` visibility anyway.

So to keep the same pattern you see in other publications, where the default is always last, we have the following in place which might be tiny bit neater.. 
Move it instead to line 242
```js
  // Everyone else gets the standard, visible products and variants
  selector = Object.assign(selector, {
    isVisible: true,
    $or: [
      { type: "simple" },
      { type: "variant" }
    ],
    isDeleted: { $in: [null, false] }
  });
```
Hope I'm not forgetting something, but I think that's it.